app/services/update_location.rb

```diff
@@ -17,12 +17,14 @@ def update!(params)
   private
 
   def create!(params, version = 1)
+    attr = location ? location.attributes.except('id', 'created_at', 'updated_at') : {}
```

Comment:
This feels a bit fragile and difficult to parse.

Should `version` be part of the params?
Should plucking the properties you need out of `location` be put into it's own method?